commands/build-image.ts

```diff
@@ -72,27 +74,33 @@ export async function buildImage(dockerFileUri?: vscode.Uri): Promise<void> {
         return;
     }
 
-    const uri: Item = await resolveImageItem(folder, dockerFileUri);
-    if (!uri) { return; }
+    while (!dockerFileItem) {
+        let resolvedItem: Item | undefined = await resolveImageItem(workspaceFolder, dockerFileUri);
+        if (resolvedItem) {
+            dockerFileItem = resolvedItem;
+        } else {
+            let msg = "Couldn't find a Dockerfile in your workspace. Would you like to add Docker files to the workspace?";
+            let response = await ext.ui.showWarningMessage(msg, DialogResponses.yes, DialogResponses.no);
+            if (response === DialogResponses.no) {
+                actionContext.properties.cancelStep = msg;
+                throw new UserCancelledError();
+            } else {
+                await vscode.commands.executeCommand('vscode-docker.configure');
+                // Try again
+            }
+        }
+    }
 
-    let contextPath: string = uri.path;
+    let contextPath: string = dockerFileItem.folderPath;
     if (defaultContextPath && defaultContextPath !== '') {
         contextPath = defaultContextPath;
     }
 
+    // Get imageName based on name of subfolder containing the Dockerfile, or else workspacefolder
     let imageName: string;
-    if (process.platform === 'win32') {
```

Comment:
omg thank you for fixing this to use `path.basename` - looks so much cleaner 😂

Reply:
:-)